parquet/src/main/java/org/apache/iceberg/parquet/ParquetWriter.java

```diff
@@ -185,9 +190,17 @@ public List<Long> splitOffsets() {
     return null;
   }
 
+  private long estimateBufferSize() {
+    if (totalRowGroupSize == 0 || totalBufferSize == 0) {
```

Comment:
Should this also fallback to `writeStore.getBufferedSize()` when `currentBufferSize == 0`? Likely it'd be `0` since I don't see other case, but I feel that'd make it consistent on the fallbacks.